core/evancore.vue.libs/src/components/forms/control/control.ts

```diff
@@ -99,11 +99,11 @@ export default class ControlComponent extends mixins(EvanComponent) {
   /**
    * Determines if the current field is required. If not, show a optional hint.
    */
-  isRequired() {
+  isRequired(): boolean {
     if (this.required && typeof this.required === 'function') {
       return this.required();
-    } else {
-      return this.required;
     }
+
+    return this.required as boolean;
```

Comment:
Use this code snippet instead :)
```
  /**
   * Is the current field is required? If not, a optional hint will be displayed
   */
  @Prop() required: boolean | (() => boolean);

  /**
   * Determines if the current field is required. If not, show a optional hint.
   */
  isRequired(): boolean {
    if (this.isFunc(this.required)) {
      return this.required();
    }

    return this.required;
  }

  isFunc(val: Function | boolean): val is Function {
    if (val) {
      return typeof val === 'function';
    }
    return false;
  }
```
More Info here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

I think we shouldn't mix primitive types with functions in the future to make the code easier to reason about.

Reply:
Crazy stuff. Thx!